apps/emqx_connector/src/emqx_connector_http.erl

```diff
@@ -366,7 +370,11 @@ on_query_async(
     ?TRACE(
         "QUERY_ASYNC",
         "http_connector_received",
-        #{request => Request, connector => InstId, state => State}
+        #{
+            request => emqx_misc:redact(Request),
+            connector => InstId,
+            state => emqx_misc:redact(State)
```

Comment:
I think we should rather use `emqx_misc:redact/2` here and use a custom `Checker` that adds support for those http headers rather then adding that check to the generic `is_sensitive_key/1`.
But probably the latter should also be exported from `emqx_misc`, then you could use it like
`emqx_misc:redacted(State, fun http_header_checker/1)` and 
```http_header_checker(Key) ->
   case emqx_misc:is_sensitive_key(Key) of 
     false -> 
      ...
      true -> 
           true
   end.
   ```

Reply:
Yes, I did something similar [here](https://github.com/emqx/emqx/pull/9840/files#diff-9c8db1387f528a80196a9d0b371694e51d55ebb27bfa8f960abefc1ed05634c6R341).

Reply:
Yes, I think this makes sense unless sensitive data also have to be redacted at a higher level. For example, if the connector crashes, then maybe the state of the connector is logged in another module. I will try to investigate a little bit more how this works before I refactor to use an module internal is_sensitive_key function.

Reply:
@olcai @sstrigler thanks I forced pushed a new commit where the redacting is localized.